packages/logging/src/logging-and-telemetry.ts

```diff
@@ -125,11 +92,11 @@ export class LoggingAndTelemetry implements MongoshLoggingAndTelemetry {
   private isBufferingTelemetryEvents = false;
 
   private deviceId: string | undefined;
-  /** @internal */
+
+  /** @internal Used for awaiting the telemetry setup in tests. */
   public setupTelemetryPromise: Promise<void> = Promise.resolve();
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  private resolveDeviceId: (value: string) => void = () => {};
+  private readonly telemetrySetup: AbortController = new AbortController();
```

Comment:
not a huge fan of this name, maybe `deviceIdResolution` though I was thinking this would be shared by the entire setup. 
I could also call it `abortController` but seems like a implementation detail.